pkg/driver/driver.go

```diff
@@ -43,56 +47,43 @@ func (c RetryConnector) Connect(ctx context.Context) (driver.Conn, error) {
 				telemetry.UpdateCurrentDbConnErr(err)
 
 				if lastErr == nil || err.Error() != lastErr.Error() {
-					c.driver.Logger.Warnw("Can't connect to database. Retrying", zap.Error(err))
+					c.Logger.Warnw("Can't connect to database. Retrying", zap.Error(err))
 				}
 			},
 			OnSuccess: func(elapsed time.Duration, attempt uint64, _ error) {
 				telemetry.UpdateCurrentDbConnErr(nil)
 
 				if attempt > 0 {
-					c.driver.Logger.Infow("Reconnected to database",
+					c.Logger.Infow("Reconnected to database",
 						zap.Duration("after", elapsed), zap.Uint64("attempts", attempt+1))
 				}
 			},
 		},
 	), "can't connect to database")
+
+	if err == nil && c.InitConn != nil {
+		err = c.InitConn(ctx, conn)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
What happens if this returns something like a network error? Shouldn't this happen within the retry just above?